pages/content/amp-dev/documentation/guides-and-tutorials/develop/custom-javascript.md

```diff
@@ -0,0 +1,139 @@
+---
+$title: Include Custom JavaScript in AMP Pages
+$order: 7
+formats:
+  - websites
+author: CrystalOnScript
+contributors:
+  - fstanis
+---
+
+AMP strives to provide a consistently good experience to all users across the web by encouraging the use of high-functioning and seamless components that are ready to go out of the box. 
+
+Some web experiences require a high amount of customization that go beyond the state binding capabilities of [`amp-bind`](https://amp.dev/documentation/components/reference/amp-bind.html?format=websites) and the dynamic data retrieval and templating functionality of [`amp-list`](https://amp.dev/documentation/components/reference/amp-list.html?format=websites) and [`amp-mustache`](https://amp.dev/documentation/components/reference/amp-mustache.html?format=websites). For those one-of-a-kind cases, AMP has created [`<amp-script>`](https://amp.dev/documentation/components/reference/amp-script.html?format=websites), a component that allows the use of arbitrary JavaScript on your AMP page without affecting the page's overall performance.
+
+
+# Inserting custom JavaScript 
+
+AMP pages support custom JavaScript through the `<amp-script>` component. 
+
+
+```html
+<!doctype html>
+<html ⚡>
+<head>
+  ...
+  <script async custom-element="amp-script" src="https://cdn.ampproject.org/v0/amp-script-0.1.js"></script>
+<body>  
+  ...
+  <amp-script layout="container" src="https://example.com/myfile.js">
+    <p>Initial content that can be modified from JavaScript</p>
+  </amp-script>
+  ...
+</body>
+</html>
+
+```
+
+
+The `<amp-script>` component registers a [Web Worker](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API) to run on a separate thread than the main page. The Web Worker is given its own copy of the DOM through `amp-script` use of [Worker DOM](https://github.com/ampproject/worker-dom). This allows the Web Worker to use JavaScript libraries, such as [React](https://reactjs.org/) and [jQuery](https://jquery.com/), without modification. 
+
+The `amp-script` component sends messages between the Web Worker thread and the main thread, causing any changes the user makes on the main DOM to be echoed on the Web Worker's false DOM. In turn, the Web Worker can then update the false DOM, which is reflected on the main DOM.
```

Comment:
I'd recommend leaving out the mention of echoing changes from the main thread to the Worker.

Reply:
seems like a potentially important detail, so leaving it in for now, but I'll let @CrystalOnScript make the call what to do here when she's less busy.